contracts/src/main/kotlin/com/r3/developers/apples/states/AppleStamp.kt

```diff
@@ -0,0 +1,19 @@
+package com.r3.developers.apples.states
+
+import com.r3.developers.apples.contracts.AppleStampContract
+import net.corda.v5.ledger.utxo.BelongsToContract
+import net.corda.v5.ledger.utxo.ContractState
+import java.security.PublicKey
+import java.util.*
+
+@BelongsToContract(AppleStampContract::class)
+class AppleStamp(
+    val id: UUID,
+    val stampDesc: String,
+    val issuer: PublicKey,
+    val holder: PublicKey,
+    private val participants: List<PublicKey>
```

Comment:
You should never pass `participants` as a constructor argument (it's bad practice because you can then just pass in whoever you want). You should pass in well defined public keys (i.e. issuer, owner, debtor, creditor, etc) for _possible_ participants, and then let `getParticipants` compute which keys should be participants.

Reply:
The best practice explanation makes sense, thanks!

Captured and will be done in https://r3-cev.atlassian.net/browse/CORE-15314 and https://r3-cev.atlassian.net/browse/CORE-15315

For now:
- like most of my responses to the other comments, I will follow the same practice/standard the Apples tutorial gives me so that people who follow the guides can see line-by-line the exact same thing expected of them.
- in this case: [the code in the guide](https://docs.r3.com/en/platform/corda/5.0/developing-applications/basic-cordapp/state.html#add-the-required-properties) is what I'll follow

For the future:
- I will make another PR to update the apples tutorial solution to be written in the best practices we decide to go forward with.